scenario/smartsearch.go

```diff
@@ -0,0 +1,350 @@
+		sessionState.LogEntry.LogDebugf(`search%d text %s becomes search terms: %s`, id, quote(searchText), quoteList(searchTerms))
+	}
+
+	sessionState.QueueRequest(func(ctx context.Context) error {
+		searchSuggestionResult, err := appDoc.SearchSuggest(
+			ctx,
+			&enigma.SearchCombinationOptions{},
+			searchTerms,
+		)
+		logSearchSuggestionResult(sessionState.LogEntry, id, searchSuggestionResult)
+		return err
+	}, actionState, true, "SearchSuggest call failed")
+
+	sessionState.QueueRequest(func(ctx context.Context) error {
+		searchResult, err := appDoc.SearchResults(
+			ctx,
+			&enigma.SearchCombinationOptions{
+				Context:      "CurrentSelections",
+				CharEncoding: "Utf16",
+			},
+			searchTerms,
+			&enigma.SearchPage{
+				Offset: 0,
+				Count:  5,
+				GroupOptions: []*enigma.SearchGroupOptions{
+					{
+						GroupType: "DatasetType",
+						Offset:    0,
+						Count:     intPtr(-1),
+					},
+				},
+				GroupItemOptions: []*enigma.SearchGroupItemOptions{{
+					GroupItemType: "Field",
+					Offset:        0,
+					Count:         intPtr(5),
+				}},
+			},
+		)
+		logSearchResult(sessionState.LogEntry, id, searchResult)
+		return err
+	}, actionState, true, "SearchResults call failed")
+}
+
+func newSearchTextChunks(randomizer helpers.Randomizer, searchText string, pasteSearchText bool) (searchTextChunks, error) {
+	if pasteSearchText {
+		return searchTextChunks{
+			searchTextChunk{
+				Text:                    searchText,
+				PostTypingThinkDuration: 0,
+				TypingDuration:          0,
+			},
+		}, nil
+	}
+
+	const typeOneCharDuration = 300 * time.Millisecond
+	const minPostTypingDuration = 700 * time.Millisecond
+	const maxPostTypingDuration = 1300 * time.Millisecond
+
+	chunks := searchTextChunks{}
+	currentStart := 0
+	for currentEnd := 1; currentEnd < len(searchText); currentEnd++ {
+		randInt, err := randomizer.RandWeightedInt([]int{1, 9})
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to randomize search text chunk")
+
+		}
+		shallSplit := randInt == 0
+		postTypingDuration, err := randomizer.RandDuration(minPostTypingDuration, maxPostTypingDuration)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to randomize search text chunk post typing duration")
+		}
+		if shallSplit {
+			textChunk := searchText[currentStart:currentEnd]
+			currentStart = currentEnd
+			chunks = append(chunks, searchTextChunk{
+				Text:                    textChunk,
+				PostTypingThinkDuration: postTypingDuration,
+				TypingDuration:          time.Duration(len(textChunk)) * typeOneCharDuration,
+			})
+		}
+	}
+	textChunk := searchText[currentStart:]
+	chunks = append(chunks, searchTextChunk{
+		Text:                    textChunk,
+		PostTypingThinkDuration: 0,
+		TypingDuration:          time.Duration(len(textChunk)) * typeOneCharDuration,
+	})
+	return chunks, nil
+
+}
+
+func (chunks searchTextChunks) simulate() <-chan string {
+	textChan := make(chan string, len(chunks))
+	go func() {
+		currentText := ""
```

Comment:
add function to recover panic and log error on actions state in the go routine. (there's a helper function for recover)